common/http.go

```diff
@@ -0,0 +1,54 @@
+package common
+
+import (
+	"net/http"
+	"time"
+)
+
+type logResponseWriter struct {
+	http.ResponseWriter
+	// As it is stated in the net/http.ResponseWriter docs, the status is set
+	// either by WriteHeader or by Write if the former has not been called.
+	// In the latter case, the status is being set to http.StatusOK.
+	status int
+}
+
+func (l *logResponseWriter) WriteHeader(status int) {
+	l.ResponseWriter.WriteHeader(status)
+	l.status = status
+}
+
+// NB: Required by http.ResponseWriters which implement http.CloseNotifier.
+func (l *logResponseWriter) CloseNotify() <-chan bool {
+	return l.ResponseWriter.(http.CloseNotifier).CloseNotify()
+}
+
+type logHandler struct {
+	handler http.Handler
+}
+
+// LogHandler creates a http.Handler for logging HTTP requests. The handler
+// wraps a given handler h.
+//
+// A format of the logging is the following:
+// $METHOD $REQUEST_URI $STATUS_CODE $DURATION
+func LogHandler(h http.Handler) http.Handler {
+	return &logHandler{handler: h}
+}
+
+// ServeHTTP implements the http.Handler interface.
+func (l *logHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	writerWrapper := &logResponseWriter{
+		ResponseWriter: w,
+		status:         http.StatusOK,
+	}
+
+	start := time.Now()
+	l.handler.ServeHTTP(writerWrapper, r)
+	duration := time.Now().Sub(start)
```

Comment:
In general, I find it useful when debugging failures happening in long chains of calls due to timeouts / timing issues. Also, it can give us a better idea on how long some requests are taking.